plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftSplitManager.java

```
@@ -0,0 +1,202 @@
+        }
+        ConnectorSplitSource fallbackSplitSource = new FixedSplitSource(new JdbcSplit(Optional.empty()));
+        if (!isUnloadEnabled(session)) {
+            return fallbackSplitSource;
+        }
+        JdbcTableHandle jdbcTable = (JdbcTableHandle) table;
+        JdbcTableHandle jdbcTableHandle = dynamicFilteringEnabled(session) ? jdbcTable.intersectedWithConstraint(dynamicFilter.getCurrentPredicate()) : jdbcTable;
+        List<JdbcColumnHandle> columns = jdbcTableHandle.getColumns()
+                .orElseGet(() -> jdbcClient.getColumns(
+                        session,
+                        jdbcTableHandle.getRequiredNamedRelation().getSchemaTableName(),
+                        jdbcTableHandle.getRequiredNamedRelation().getRemoteTableName()));
+
+        if (!isUnloadSupported(jdbcTable, columns)) {
+            log.debug("Unsupported query shape detected. Falling back to using JDBC");
+            return fallbackSplitSource;
+        }
+
+        Connection connection;
+        PreparedStatement statement;
+        String unloadOutputPath = unloadLocation.orElseThrow() + "/" + session.getQueryId() + "-" + UUID.randomUUID() + "/";
+        try {
+            connection = jdbcClient.getConnection(session);
+            String redshiftSelectSql = getRedshiftSelectSql(session, connection, jdbcTableHandle, columns);
+
+            // Query containing \\b is unsupported with unload command. See https://github.com/aws/amazon-redshift-jdbc-driver/issues/124
+            if (redshiftSelectSql.contains("\\b")) {
+                log.debug("Unload query contains unsupported characters. Falling back to using JDBC");
+                return fallbackSplitSource;
+            }
+            statement = buildUnloadSql(session, connection, columns, redshiftSelectSql, unloadOutputPath);
+        }
+        catch (SQLException e) {
+            if (e instanceof RedshiftException) {
```

This exception seems broad. There are many subclasses maybe we should just throw an exception. https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/redshift/model/RedshiftException.html. Can we narrow down the condition?

---

I changed the code to throw instead.